mscore/startupWizard.cpp

```diff
@@ -0,0 +1,108 @@
+      {QT_TRANSLATE_NOOP("keyboard-layout", "T2 - German"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Swiss German"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Belgian French"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Canadian French"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "French French"), ":/data/shortcuts_AZERTY.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Swiss French"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Spanish"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Italian"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Brazilian Portuguese ABNT"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Portuguese"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "US - International"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Dutch"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Korean - Dubeolsik"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Korean - Sebeolsik"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Other"), ":/data/shortcuts.xml"},
+      };
+    
+const static QMap<QString, QString> langToLayout = {
+      {"en", QT_TRANSLATE_NOOP("keyboard-layout", "US-QWERTY")},
+      {"ja", QT_TRANSLATE_NOOP("keyboard-layout", "JIS - Standard Japanese")},
+      {"de", QT_TRANSLATE_NOOP("keyboard-layout", "T1 - Standard German")},
+      {"fr", QT_TRANSLATE_NOOP("keyboard-layout", "French French")},
+      {"es", QT_TRANSLATE_NOOP("keyboard-layout", "Spanish")},
+      {"it", QT_TRANSLATE_NOOP("keyboard-layout", "Italian")},
+      {"pt", QT_TRANSLATE_NOOP("keyboard-layout", "Portuguese")},
+      {"nl", QT_TRANSLATE_NOOP("keyboard-layout", "Dutch")},
+      {"ko", QT_TRANSLATE_NOOP("keyboard-layout", "Korean - Dubeolsik")}
+      };
+
+QString StartupWizard::startup_wizard()
+      {
+      QWizardPage *layoutInput = new QWizardPage;
+      layoutInput->setTitle("Keyboard Layout");
+      QLabel *label = new QLabel(QObject::tr("Enter your keyboard layout"));
+      QStringList layoutList;
+      QList<QString> keyboardLayouts = layoutToShortcut.keys();
+      for (auto layout : keyboardLayouts) {
+            layoutList.append(qApp->translate("keyboard-layout", layout.toStdString().c_str()));
+      }
+      QComboBox *key_Layouts = new QComboBox();
+      key_Layouts->addItems(layoutList);
+      QVBoxLayout *layout = new QVBoxLayout();
+      layout->addWidget(label);
+      layout->addWidget(key_Layouts);
+      layoutInput->setLayout(layout);
+      QString lang = Ms::mscore->getLocaleISOCode().left(2);
+      QString bestLayout;
+      try   {
+            bestLayout = langToLayout.value(lang);
```

Comment:
We don't use try catch or in very particular cases. Just get the value and pass the default http://doc.qt.io/qt-5/qmap.html#value